src/main/java/com/dev/movietheatre/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,52 @@
+package com.dev.movietheatre.service.impl;
+
+import com.dev.movietheatre.dao.ShoppingCartDao;
+import com.dev.movietheatre.dao.TicketDao;
+import com.dev.movietheatre.lib.Inject;
+import com.dev.movietheatre.lib.Service;
+import com.dev.movietheatre.model.MovieSession;
+import com.dev.movietheatre.model.ShoppingCart;
+import com.dev.movietheatre.model.Ticket;
+import com.dev.movietheatre.model.User;
+import com.dev.movietheatre.service.ShoppingCartService;
+import java.util.List;
+
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+    @Inject
+    private TicketDao ticketDao;
+
+    @Override
+    public void addSession(MovieSession movieSession, User user) {
+        Ticket ticket = new Ticket();
+        ticket.setMovieSession(movieSession);
+        ticket.setUser(user);
+        ticketDao.add(ticket);
+        ShoppingCart shoppingCartByUser = shoppingCartDao.getByUser(user);
+        List<Ticket> tickets = shoppingCartByUser.getTickets();
+        tickets.add(ticket);
+        shoppingCartDao.update(shoppingCartByUser);
```

Comment:
You add the ticket to the List, but I don't sure what tickets are updating in the shopping cart. And you can simplify your code: 
```suggestion
        shoppingCartByUser.getTickets().add(ticket);
        shoppingCartDao.update(shoppingCartByUser);
```

Reply:
done